core/eolearn/tests/test_core_tasks.py

```diff
@@ -376,6 +401,53 @@ def test_map_features(test_eopatch):
         MapFeatureTask(f_in, f_out)
 
 
+@pytest.mark.parametrize(
+    "input_features, output_features, map_function, kwargs",
+    [
+        (
+            {FeatureType.DATA: ["CLP", "bands"]},
+            {FeatureType.DATA_TIMELESS: ["CLP_max", "bands_max"]},
+            np.max,
+            {"axis": -1},
+        ),
+        (
+            {FeatureType.DATA: ["CLP", "bands"]},
+            {FeatureType.DATA: ["CLP_+3", "bands_+3"]},
+            lambda x, a: x + a,
+            {"a": 3},
+        ),
+    ],
+)
+def test_map_kwargs(
+    input_features: FeaturesSpecification,
+    output_features: FeaturesSpecification,
+    map_function: Callable,
+    kwargs: Dict[str, Any],
+    patch: EOPatch,
+) -> None:
+    mapped_patch = MapFeatureTask(input_features, output_features, map_function, **kwargs)(patch)
+
+    for in_feature, out_feature in zip(parse_features(input_features), parse_features(output_features)):
+        expected_output = map_function(mapped_patch[in_feature], **kwargs)
+        assert_array_equal(mapped_patch[out_feature], expected_output)
+
+
+@pytest.mark.parametrize(
+    "input_feature, kwargs",
+    [
+        ((FeatureType.DATA, "bands"), {"axis": -1, "name": "fun_name", "bands": [4, 3, 2]}),
+    ],
+)
+def test_map_kwargs_passing(input_feature: FeatureSpec, kwargs: Dict[str, Any], patch: EOPatch) -> None:
+    def kwargs_map(data, *, some=3, **kwargs) -> tuple:
+        return some, kwargs
+
+    mapped_patch = MapFeatureTask(input_feature, (FeatureType.META_INFO, "kwargs"), kwargs_map, **kwargs)(patch)
+
+    expected_output = kwargs_map(mapped_patch[input_feature], **kwargs)
+    assert_array_equal(mapped_patch[(FeatureType.META_INFO, "kwargs")], expected_output)
```

Comment:
these two should be equal, not array equal.

it also makes no sense to keep this and the above test. This was suggested as a simplification of `test_map_kwargs`. 
If you do not like this test, then I believe you can merge `test_map` and `test_map_kwargs` since the tests are identical except for the `kwargs` parameter

Reply:
> these two should be equal, not array equal.

Done.

> it also makes no sense to keep this and the above test. This was suggested as a simplification of test_map_kwargs.
If you do not like this test, then I believe you can merge test_map and test_map_kwargs since the tests are identical except for the kwargs parameter

Agree, i think that "new" test kinda do not test if parameters were appealed to map function just if they were passed.  
Kinda done, parametrization look a bit "ugly".

Reply:
the joint tests are ok enough, the parametrization is large but i guess it has to be.